### DISCUSSION TITLE
Evolving the magnetic field question

### DISCUSSION THREAD

**Discussion by kurtsansom** [PrincetonUniversity/athena#408] [Q&A]
I have model where I have initialized the density, energy and magnetic field
and then specify the inner boundaries for velocity, energy and magnetic field and set the outer boundaries to `outflow`

It appears that the magnetic field does not change as the simulation evolves. I tried disabling the magnetic field initialization and only setting the inner boundary magnetic field and leave the outer boundary to outflow. The magnetic field is different but it doesn't appear to evolve in time either. I believe I am missing something but I don't know what.

Since this is a custom pgen problem do I need to evolving the magnetic field in the problem generator code?
or a flag that is missing.

magnetic fields are enabled, hlld solver and xorder=3 with rk4 timestepping.

What is weird to me is that the bcc2 direction is evolving but not the bcc1 direction.

**Comment by tomidakn** [comment]
I cannot identify your problem from what you described, but I speculate you were solving one dimensional problem. Here one dimensional can be either truly 1D (nx2=nx3=1 in the code) or the input is one dimensional (uniform in the x2 and x3 directions). In this case, from the induction equation, Bx stays unchanged no matter what profile you have in the x1 direction (and at the same time Bx has to be spatially constant to satisfy the divB=0 constraint). This is a basic consequence of MHD, so I would recommend you to read a textbook of MHD if you are not familiar.

The CT algorithm used in Athena++ is very sensitive to divB error. So you have to use a vector potential to initialize the magnetic field except for trivial cases like uniform fields. Please consult the documentation and the method paper.

**Reply by kayarre** [reply]
@tomidakn Thank you so much for taking the time to explain. I do not have much formal MHD training and appreciate your patience with my questions.  (if you have a good MHD book recommendation I would appreciate it)

I am trying to model the background solar wind expanding from the sun starting from about 0.1 AU. It is currently modeled in 2D cylindrical coordinates. I think your explanation makes it somewhat more clear what I am missing.

The magnetic field is initialized to that of a Parker spiral. 

I interpret the explanation as once the magnetic field is initialized I should not necessarily expect it to change in the x1 direction.

**Reply by tomidakn** [reply]
If your initial condition has non-trivial structure in 2D, it should evolve.  The Parker Spiral is a steady state solution, but usually it is very hard for any MHD code to maintain the exact equilibrium, so you *should see some change in the x1 direction*.  (note: Bz should remain zero if it is initially zero everywhere.) So I'm afraid something is wrong but I cannot point out what is wrong in your case.

**Reply by kurtsansom** [reply]
I think the potential problem is not accounting for orbital advection. Will investigate. Thank you @tomidakn for your input.
